File: app/src/main/java/com/funckyhacker/fileexplorer/di/SearchComponent.java
package com.funckyhacker.fileexplorer.di;

import com.funckyhacker.fileexplorer.view.search.SearchActivity;
import dagger.Subcomponent;
import dagger.android.AndroidInjector;

@Subcomponent
public interface SearchComponent extends AndroidInjector<SearchActivity>{
  @Subcomponent.Builder
  abstract class Builder extends AndroidInjector.Builder<SearchActivity> {
  }
}

File: app/src/main/java/com/funckyhacker/fileexplorer/di/SearchComponent.kt
package com.funckyhacker.fileexplorer.di

import com.funckyhacker.fileexplorer.view.search.SearchActivity
import dagger.Subcomponent
import dagger.android.AndroidInjector

@Subcomponent
interface SearchComponent : AndroidInjector<SearchActivity> {
    @Subcomponent.Builder
    abstract class Builder : AndroidInjector.Builder<SearchActivity>()
}
